api/app/hfi/hfi.py

```diff
@@ -0,0 +1,118 @@
+
+        # Take only the number of days requested for
+        prep_week_dailies = area_dailies_by_date[:num_prep_days]
+
+        # Initialize with defaults if empty
+        if area.name not in planning_area_fire_starts:
+            planning_area_fire_starts[area.name] = [lowest_fire_starts for _ in range(num_prep_days)]
+
+        daily_results: List[DailyResult] = []
+        all_dailies_valid: bool = True
+        for index, prep_day_dailies in enumerate(prep_week_dailies):
+            daily_fire_starts: FireStartRange = planning_area_fire_starts[area.name][index]
+            mean_intensity_group = calculate_mean_intensity(prep_day_dailies)
+            prep_level = calculate_prep_level(mean_intensity_group, daily_fire_starts)
+            validated_dailies: List[ValidatedStationDaily] = list(
+                map(validate_station_daily, prep_day_dailies))
+            all_dailies_valid = all(map(lambda validated_daily: (validated_daily.valid), validated_dailies))
+            daily_result: DailyResult = DailyResult(
+                dateISO=prep_day_dailies[0].date.isoformat(),
+                dailies=validated_dailies,
+                fire_starts=daily_fire_starts,
+                mean_intensity_group=mean_intensity_group,
+                prep_level=prep_level)
+            daily_results.append(daily_result)
+
+        highest_daily_intensity_group = calculate_max_intensity_group(
+            list(map(lambda daily_result: (daily_result.mean_intensity_group), daily_results)))
+
+        mean_prep_level = calculate_mean_prep_level(
+            list(map(lambda daily_result: (daily_result.prep_level), daily_results)))
+
+        planning_area_to_dailies[area.name] = PlanningAreaResult(
+            all_dailies_valid=all_dailies_valid,
+            highest_daily_intensity_group=highest_daily_intensity_group,
+            mean_prep_level=mean_prep_level,
+            daily_results=daily_results)
+
+    return planning_area_to_dailies
+
+
+def calculate_max_intensity_group(mean_intensity_groups: List[Optional[float]]):
+    """ Returns the highest intensity group from a list of values """
+    valid_mean_intensity_groups = list(filter(None, mean_intensity_groups))
+    return None if len(valid_mean_intensity_groups) == 0 else max(mean_intensity_groups)
+
+
+def calculate_mean_prep_level(prep_levels: List[Optional[float]]):
+    """ Returns the mean prep level from a list of values """
+    valid_prep_levels = list(filter(None, prep_levels))
+    return None if len(valid_prep_levels) == 0 else mean(valid_prep_levels)
+
+
+def calculate_mean_intensity(dailies: List[StationDaily]):
+    """ Returns the mean intensity group from a list of values """
+    intensity_groups = list(map(lambda daily: (daily.intensity_group), dailies))
+    valid_intensity_groups = list(filter(None, intensity_groups))
+    return None if len(valid_intensity_groups) == 0 else \
+        round((10 * sum(valid_intensity_groups) / len(valid_intensity_groups))) / 10
```

Comment:
couldn't you just use `mean(valid_intensity_groups)` like you've done for mean prep level?

Reply:
I've been wondering this too and think it makes sense. This code is verbatim what's on the frontend and I cannot remember why it is this way.

Reply:
Made the change and the tests are happy, just pushed it.